docs/topic-guides/kotlin/suggest.md

```diff
@@ -0,0 +1,97 @@
+# Suggest
+
+The API for the `SuggestStore` can be found in the [MozillaComponents Kotlin documentation](https://mozilla.github.io/application-services/kotlin/kotlin-components-docs/mozilla.appservices.suggest/-suggest-store/index.html).
+
+> The `SuggestStore` is a synchronous, which needs to be wrapped in the asynchronous primitive of the target language you are using it in.
+
+## Setting up the store
+
+You need to import the following primitives to work with the `SuggestStore` (these come from the generated `suggest.swift` file, produced by `uniffi`):
+
+```kotlin
+import mozilla.appservices.remotesettings.RemoteSettingsServer
+import mozilla.appservices.suggest.SuggestApiException
+import mozilla.appservices.suggest.SuggestIngestionConstraints
+import mozilla.appservices.suggest.SuggestStore
+import mozilla.appservices.suggest.SuggestStoreBuilder
+import mozilla.appservices.suggest.Suggestion
+import mozilla.appservices.suggest.SuggestionQuery
+```
+
+On start up of your application, you create a `SuggestStore` (as a singleton). You do this via the `SuggestStoreBuilder`, which returns a `SuggestStore`:
+
+```kotlin
+internal val store: Lazy<SuggestStore> = lazy {
+    SuggestStoreBuilder()
+        .dataPath(context.getDatabasePath(DATABASE_NAME).absolutePath)
+        .remoteSettingsServer(remoteSettingsServer)
+        .build()
+```
+
+In addition to the `dataPath`, which is the path (the SQLite location) where you store your suggestions, you need a `remoteSettingsServer`, which is the URL to the instance of `RemoteSettings` you will be using (e.g. `https://remote-settings-dev.allizom.org`)
+
+## Ingesting suggestions
+
+We have to handle three different phases of the ingesting suggestions: The initial one on starting up the application, periodically in the background, and aborting an ongoing ingestion.
+
+### On Start Up
+Ingest with `SuggestIngestionConstraints(emptyOnly=true)` shortly after each startup. This ensures we have something in the DB on the first run and also after upgrades where we often will clear the DB to start from scratch.
+
+```kotlin
+store.value.ingest(SuggestIngestionConstraints(emptyOnly = true))
+```
+
+### Periodically
+
+Ingest with `SuggestIngestionConstraints(emptyOnly=false)` on regular schedule (like once a day).
+
+Example:
+
+```kotlin
+store.value.ingest(SuggestIngestionConstraints(emptyOnly = false))
+```
+
+The defintion of `SuggestIngestionConstraints`:
```

Comment:
I'd rather this be more example-focused.   How about saying something like:
  - Starting with "SuggestIngestionConstraints" is used to control what gets ingested (link to the API docs)
  - Use the `providers` field to limit ingestion by provider type (show an example where you're only ingesting a couple providers.   `AMP_MOBILE`, `WIKIPEDIA`, and `WEATHER` are a reasonable combination.
  - Use the `providerConstraints` field to add additional constraints, currently this is only used for exposure suggestions.  (I don't have a great example for this field. @0c0w3 do you have any suggestions?)